### DISCUSSION TITLE
SSL Fingerprint: Disabled

### DISCUSSION THREAD

**Discussion by RUXYSDigest** [debauchee/barrier#1682] [Q&A] [answered]
I have installed the Windows 10 and Linux versions of Barrier via BarrierSetup-2.4.0-release.exe for Windows and Snap Install(ver. 2.4.0-release-ac5a1bfd). Both show the Barrier software setting SSL "Fingerprint: Disabled" with no option of enabling the setting. I have viewed several tutorials on youtube on how to get it up and running. I am able to follow along and input the needed info, however, most if not all of the videos show the SSL Fingerprint are with some type of certificate address or something.
On Windows 10 I've switched off McAfee, and Ubuntu 20.04 UFW settings are good and still don't have a clue as to what is wrong. In addition, I am currently on Google Fiber(5G) as my internet provider. I have my suspicions but don't want to jump to conclusions.
Please help. Would love to get this up and running as easily as shown in the videos. Thanks in advance.
UPDATE: I found where to enable the SSL and both were checked (while still showing SSL Fingerprint: Disabled). Unchecked them both, stopped Barriers, quit & restart. Still unable to connect. Clicked them both back on (and added Require client certificate for good measure) and still displaying SSL Fingerprint: Disabled.
UPDATE2: Got into Google Home app (Port Forward settings) opened up 24800 on the server computer. Went into Ubuntu's UFW settings and opened up port 24800 with no results. I am, however, getting an "ssl certificate doesn't exist: C:\User..." on the Win10 side from the Barrier's Logs. And still, the "SSL Fingerprint: Disabled".
[2022-06-01T10:48:06] ERROR: ssl certificate doesn't exist: C:\Users\Joh****epp\AppData\Local\Barrier\SSL\Barrier.pem
[2022-06-01T10:48:06] ERROR: could not load client certificates
[2022-06-01T10:48:06] ERROR: ssl error occurred (system call failure)
[2022-06-01T10:48:06] ERROR: failed to connect secure socket
[2022-06-01T10:48:21] WARNING: failed to connect to server: Timed out
[2022-06-01T10:48:22] NOTE: connecting to '192.867.53.09': 192.867.53.09:24800
UPDATE3: My 2 computers are wired to Google Fibers (hockey puck) wifi router via a 5-port Desktop switch. Will switch both computers to wifi and see if that has any effect... yep, no dice. I even tried enabling the Auto config w/ Bonjour on the windows side. Looking for an SSL certificate and the SSL Fingerprint: Disabled; even after multiple uninstalls, re-installs.

**Comment by RUXYSDigest** [comment, marked as answer]
Goto https://stackoverflow.com/questions/67343804/error-ssl-certificate-doesnt-exist-home-rsvay-snap-barrier-kvm-2-local-shar
The paths may not align but if you follow the steps, you'll get it working. Win 10 & Ubuntu 20.04.4. Had to go to ~/snap/barrier/<3digit#>/.local/share/barrier/SSL and created the Fingerprints folder.

Stay in the SSL directory to run the script mentioned.

(You have to create a text file Local.txt in the Fingerprints directory but run the script from SSL).
I don't have enough points to thank you properly @ stackoverflow however, many times over... Thank U for your solution!!!

**Comment by jhgorse** [comment]
# Summary of Resolutions
Note: Windows requires https://gitforwindows.org/

## Windows 
```Powershell
# Powershell
Set-Alias openssl "C:\Program Files\Git\usr\bin\openssl.exe"
cd ~\AppData\Local\Barrier\SSL\
openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
```
```cmd
REM cmd
cd C:\Users\%USERNAME%\AppData\Local\Barrier\SSL\
openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
```

## Linux, macOS, and unix style
Pick your `BARRIER_SSL_PATH` path.
```bash
### Ubuntu 22 Apt
BARRIER_SSL_PATH=~/.local/share/barrier/SSL/
### Ubuntu Snap
BARRIER_SSL_PATH=~/snap/barrier/current/.local/share/barrier/SSL
### Fedora Snap
BARRIER_SSL_PATH=~/snap/barrier-kvm/current/.local/share/barrier/SSL/
### macOS
BARRIER_SSL_PATH=~/Library/Application Support/barrier/SSL

mkdir -p "${BARRIER_SSL_PATH}" && cd "${BARRIER_SSL_PATH}"
openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
```

Use the unix version with the correct path for your system. Find the path in the Barrier log.

2023-02-26: Git for Windows dependency added
2023-11-20: cd to improve openssl's ability to handle spaces in directory names

Cheers

Originally Posted here: https://github.com/debauchee/barrier/issues/231#issuecomment-1143791895

**Reply by onlinedev-genius** [reply]
I was able to resolve the issue with this solution:
```
# Powershell
Set-Alias openssl "C:\Program Files\Git\usr\bin\openssl.exe"
cd ~\AppData\Local\Barrier\SSL\
openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem
```
Thank you @jhgorse
